docker/Dockerfile_stack

```diff
@@ -74,22 +70,48 @@ RUN export OPENMP_VERSION=4.0.0 && \
     mkdir -p ${OPENMP_BUILD_DIR} && \
     cd ${OPENMP_BUILD_DIR} && \
     cmake \
-        -D CMAKE_C_COMPILER=clang-4.0 \
-        -D CMAKE_CXX_COMPILER=clang++-4.0 \
+        -D CMAKE_C_COMPILER=clang-${CLANG_VERSION} \
+        -D CMAKE_CXX_COMPILER=clang++-${CLANG_VERSION} \
         ${OPENMP_SOURCE_DIR} \
         && \
     make -j${NPROC} install && \
     rm -rf ${OPENMP_ARCHIVE} && \
     rm -rf ${OPENMP_BUILD_DIR} && \
     rm -rf ${OPENMP_SOURCE_DIR}
 
+# Install OpenMPI
+RUN export OPENMPI_VERSION=3.0.0 && \
+    export OPENMPI_VERSION_SHORT=3.0 && \
+    export OPENMPI_SHA1=3200b90ac01e0552c19432ffeed207a81855a7c4 && \
+    export OPENMPI_URL=https://www.open-mpi.org/software/ompi/v${OPENMPI_VERSION_SHORT}/downloads/openmpi-${OPENMPI_VERSION}.tar.gz && \
+    export OPENMPI_ARCHIVE=${PREFIX}/archive/openmpi-${OPENMPI_VERSION}.tar.gz && \
+    export OPENMPI_SOURCE_DIR=${PREFIX}/source/openmpi/${OPENMPI_VERSION} && \
+    export OPENMPI_BUILD_DIR=${PREFIX}/build/openmpi/${OPENMPI_VERSION} && \
+    export OPENMPI_INSTALL_DIR=/opt/openmpi/${OPENMPI_VERSION_SHORT} && \
+    wget --quiet ${OPENMPI_URL} --output-document=${OPENMPI_ARCHIVE} && \
+    echo "${OPENMPI_SHA1} ${OPENMPI_ARCHIVE}" | sha1sum -c && \
+    mkdir -p ${OPENMPI_SOURCE_DIR} && \
+    tar -xf ${OPENMPI_ARCHIVE} -C ${OPENMPI_SOURCE_DIR} --strip-components=1 && \
+    mkdir -p ${OPENMPI_BUILD_DIR} && \
+    cd ${OPENMPI_BUILD_DIR} && \
+    ${OPENMPI_SOURCE_DIR}/configure --prefix=${OPENMPI_INSTALL_DIR} && \
+    make -j${NPROC} install && \
+    rm -rf ${OPENMPI_ARCHIVE} && \
+    rm -rf ${OPENMPI_BUILD_DIR} && \
+    rm -rf ${OPENMPI_SOURCE_DIR}
+
+ENV OPENMPI_DIR=/opt/openmpi/3.0/bin
```

Comment:
To be more consistent with other `_DIR` env variables we set I would do `OPENMPI_DIR=/opt/openmpi/3.0`